src/Apps/Auction/Routes/ConfirmBid/index.tsx

```diff
@@ -54,6 +55,7 @@ const MAX_POLL_ATTEMPTS = 20
 
 export const ConfirmBidRoute: React.FC<ConfirmBidProps> = props => {
   let pollCount = 0
+  let registrationTracked = false
```

Comment:
probably a newbie question - but why the need to maintain state here like this?

Reply:
Also wondering this (I think it has already been explained to me in some other context but why not use `useState`?

Reply:
> why the need to maintain state here like this?

The user would stay on the same page if the user wasn't registered, tried to bid and register but the user was outbid. This way the user will be able to place another bid without a lot of distraction (and also the simplest we could implement). We want to avoid calling the `registration submitted` event more than once in such a case.

> why not use `useState`?

We don't want the whole component to get re-rendered on every internal state update so components like `<OnSubmitValidationError>` won't fire the callback too many times and React won't run unnecessary VDOM diff-ing.